libraries/outputs.rb

```diff
@@ -24,6 +24,35 @@ def self.configure_outputs(node) # rubocop:disable Metrics/PerceivedComplexity,
         else
           [CernerSplunk.my_cluster(node)]
         end.each do |(cluster, bag)|
+          if bag['indexer_discovery'] == true
+            Chef::Log.warn "Configured ['receivers'] in cluster #{cluster} will be ignored since ['indexer_discovery'] is set to true." if bag['receivers']
+
+            indexer_discovery_settings = ((bag['indexer_discovery_settings'] && bag['indexer_discovery_settings']['outputs_configs']) || {}).reject do |k, _|
+              k.start_with?('_cerner_splunk')
+            end
+            output_stanzas["indexer_discovery:#{cluster}"] = indexer_discovery_settings
+
+            if bag['master_uri'].nil? || bag['master_uri'].empty?
+              Chef::Log.warn "master_uri is missing in the cluster databag: #{cluster}"
```

Comment:
I think this would be a failure state? we would be required to use a master_uri for indexer discovery yes?

Reply:
Not setting the master_uri would result in indexer discovery not working and hence logs not getting forwarded. I thought this should be handled in the same way when `receivers` are not set. https://github.com/cerner/cerner_splunk/blob/stable/libraries/outputs.rb#L33
That was why I decided to WARN and not to FAIL.

Reply:
Hrm... yet again... who wrote this thing? Yep, I'm a clown... that apparently has been in there since before the open source version... possibly all the way back to the beginning of the project.

There is a slight difference, as here we're in a place where someone has explicitly said they want to use indexer discovery, but failed to provide the required parameter to use it... versus with the receiver bags, we're in a place where we have implied that someone should have configured something... (but also that place doesn't really make much sense to not having receivers configured there either).

Reply:
Updated to throw error and added some tests: b994c7015ea7ea3bec8b39a9d81a6e95218c45de

From our conversation yesterday, I updated the receivers check as well to throw an error instead of a warning. This will be kind of a non-passive change for those who haven't configured the receivers. But based on what we discussed yesterday that is fine because they will now configure receivers which they should.

Reply:
Above commit started failing rspec on ruby 2.1.6. Rubocop didn't like the updates that I made to fix those tests. cc8505883ed9887bec285f2f1acdd4b62fb5da64 should make rubocop and rspec happy.